addons/point_of_sale/static/src/app/models/related_models.js

```diff
@@ -492,10 +495,7 @@ export function createRelatedModels(modelDefs, modelClasses = {}, indexes = {})
             }
         }
 
-        orderedRecords[model] = orderedRecords[model].filter((rec) => rec.id !== record.id);
-
-        delete records[model][id];
-
+        records[model].delete(id);
```

Comment:
```suggestion
        delete records[model][id];
```

`records[model]` is still an object, not a map.

Reply:
oh nvm